model/shared/src/main/scala/explore/modes/SpectroscopyModesMatrix.scala

```diff
@@ -253,17 +237,25 @@ object SpectroscopyModeRow {
       ) { w =>
         import spire.std.bigDecimal._
 
-        val λr = r.wavelengthRange.toValue[BigDecimal]
-        val λm = r.minWavelength.w.micrometer
-        val Δ  = λr / TwoFactor
-        val λ  = w.micrometer
-        val λa = λ - Δ
-        val λb = λ + Δ
-        if (λa <= λm) {
-          Interval.closed(λm, λm + λr)
+        val λr     = r.wavelengthRange.toValue[BigDecimal]
+        // Range of allowed wavelength
+        val λmin   = r.minWavelength.w.micrometer
+        val λmax   = r.maxWavelength.w.micrometer
+        val Δ      = λr / TwoFactor
+        val λ      = w.micrometer
+        val λa     = λ - Δ
+        val λb     = λ + Δ
+        // if we are below min clip but shift the range
+        // same if we are above max
+        // At any event we clip at min/max
+        val (a, b) = if (λa < λmin) {
+          (λmin, λmax + λmin - λa)
```

Comment:
For some reason, I keep wanting this to be `(λmin, λmin + λr)`.  Then the else branch below would be `(λmax - λr, λmax)`.  I can't quite put together what is going on though.